src/Cameras/Inputs/freeCameraMouseWheelInput.ts

```diff
@@ -0,0 +1,373 @@
+    }
+
+    /**
+     * Set which movement axis (relative to the scene) the mouse wheel's X axis
+     * controls.
+     * @param axis The axis to be moved. Set null to clear.
+     */
+    @serialize()
+    public set wheelXMoveScene(axis: Nullable<AXIS>) {
+        if (axis === null && this._wheelXAction !== this._rotateRelative) {
+            // Attempting to clear different _wheelXAction.
+            return;
+        }
+        this._wheelXAction = this._moveScene;
+        this._wheelXActionProperty = axis;
+    }
+
+    /**
+     * Get the configured movement axis (relative to the scene) the mouse wheel's
+     * X axis controls.
+     * @returns The configured axis or null if none.
+     */
+    public get wheelXMoveScene(): Nullable<AXIS> {
+        if (this._wheelXAction !== this._moveScene) {
+            return null;
+        }
+        return this._wheelXActionProperty;
+    }
+
+    /**
+     * Set which movement axis (relative to the scene) the mouse wheel's Y axis
+     * controls.
+     * @param axis The axis to be moved. Set null to clear.
+     */
+    @serialize()
+    public set wheelYMoveScene(axis: Nullable<AXIS>) {
+        if (axis === null && this._wheelYAction !== this._rotateRelative) {
+            // Attempting to clear different _wheelYAction.
+            return;
+        }
+        this._wheelYAction = this._moveScene;
+        this._wheelYActionProperty = axis;
+    }
+
+    /**
+     * Get the configured movement axis (relative to the scene) the mouse wheel's
+     * Y axis controls.
+     * @returns The configured axis or null if none.
+     */
+    public get wheelYMoveScene(): Nullable<AXIS> {
+        if (this._wheelYAction !== this._moveScene) {
+            return null;
+        }
+        return this._wheelYActionProperty;
+    }
+
+    /**
+     * Set which movement axis (relative to the scene) the mouse wheel's Z axis
+     * controls.
+     * @param axis The axis to be moved. Set null to clear.
+     */
+    @serialize()
+    public set wheelZMoveScene(axis: Nullable<AXIS>) {
+        if (axis === null && this._wheelZAction !== this._rotateRelative) {
+            // Attempting to clear different _wheelZAction.
+            return;
+        }
+        this._wheelZAction = this._moveScene;
+        this._wheelZActionProperty = axis;
+    }
+
+    /**
+     * Get the configured movement axis (relative to the scene) the mouse wheel's
+     * Z axis controls.
+     * @returns The configured axis or null if none.
+     */
+    public get wheelZMoveScene(): Nullable<AXIS> {
+        if (this._wheelZAction !== this._moveScene) {
+            return null;
+        }
+        return this._wheelZActionProperty;
+    }
+
+    /**
+     * Called for each rendered frame.
+     */
+    public checkInputs(): void {
+        if (this._wheelDeltaX === 0 &&
+                this._wheelDeltaY === 0 &&
+                this._wheelDeltaZ == 0) {
+            return;
+        }
+
+        this._moveRelative.setAll(0);
```

Comment:
All in all I find the way you deal with the vector references a bit confusing (referencing the vectors that will be updated).

Those 3 lines are hard to get until you read the entire code. because you are resetting them, and then, technically, never set their values again. 

Maybe setting some form of an ActionType enum that will be checked during updateCamera is enough?

Reply:
Yes,
I originally had things the other way around; Each vector axis (`_moveRelative`, `_rotateRalative`, `_moveScene`) referenced the touch-wheel axis that should control it.
While this sounds more sane,
It actually lead to much more code: You could easily map a single touch-wheel axis to multiple vector axis. Checks to insure that an error was raised when that happened needed getters and setters for /every/ vector axis.

I think it's very important to get this part right though. If it needs more code, lets do that.
This is how the user interacts with this code so it has to be logical.
I think whatever API we pick should be applicable to other Inputs too. eg: Mouse/Touch input.
I rant about this part here:
https://forum.babylonjs.com/t/universal-camera-with-mouse-scroll-wheel-and-touch-pinch/14139/5

Reply:
If we do decide to leave the API as have it, I think we can fix the ambiguity in these lines with inline comments explaining what's going on.

Reply:
> Maybe setting some form of an ActionType enum that will be checked during updateCamera is enough?

I actually missed your point here on the first read. (Was too busy going off on my API rant.)
Yes, that will make things easier to understand. I'll do that.
(I'd still like a unified API for all Inputs though....)

Reply:
Hey, civilized rants are the main reason things change. :-)

I have no issue regarding the reason you are adding those three "spaces" which the user can control with the input. it was just the matter of not seeing them update in code. First thing I would do is search for `_moveRelative.set` or `_moveRelative = ???` , but the only thing i could find (outside of setters for other properties) was the `setAll(0)`. Not an issue if you look further into the code, but we are all lazy developers...

I totally get what you are saying. I have the forum page open and will answer when I get the time between PRs and refactors. 4.2 is coming up and we are all very busy trying to make the best out of this release.

Thanks for the changes! I will merge this very soon.